presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1464,7 +1478,8 @@ private static boolean meetsPartitioningRequirements(PreferredProperties preferr
         if (!preferredGlobal.getPartitioningProperties().isPresent()) {
             return !actual.isSingleNode();
         }
-        return actual.isStreamPartitionedOn(preferredGlobal.getPartitioningProperties().get().getPartitioningColumns());
+        //TODO No session here and this is only called from test code. Is this OK?
```

Comment:
If it is only used for tests - it should be fine. Please don't forget to remove this comment.